src/app/services/query.service.ts

```diff
@@ -134,15 +139,22 @@ export class QueryService {
    * Flag indicating if to relay the counts in the current step to the next
    */
   private _countsRelay: boolean;
+  /**
+   * Flag indicating if the subject selection of step 1 should be automatically
+   * saved as subject set in the backend. If true, that subject set is used as the subject constraint
+   * for step 2.
+   */
+  private _autosaveSubjectSets: boolean;
 
-
-  constructor(private resourceService: ResourceService,
+  constructor(private appConfig: AppConfig,
+              private resourceService: ResourceService,
               private treeNodeService: TreeNodeService,
               private constraintService: ConstraintService) {
     this.instantCountsUpdate_1 = false;
     this.instantCountsUpdate_2 = false;
-    this.treeNodeCountsUpdate = true;
+    this.treeNodeCountsUpdate = appConfig.getConfig('tree-node-counts-update', true);
     this.countsRelay = false;
+    this.autosaveSubjectSets = appConfig.getConfig('autosave-subject-sets', false);
```

Comment:
just a remark about the way that glowing bear is handling these parameters/flags now, there is this inconsistency: some flags are directly set in query.service, such as instantCountsUpdate_1, some flags are set via external config file such as treeNodeCountsUpdate, which may need some improvement. 

ps: omg, almost forgot that I am on holiday! :)